service_instrument.go

```diff
@@ -371,11 +410,41 @@ func (s *instrumentService) UpdateInstrument(ctx context.Context, instrument Ins
 		return err
 	}
 
+	if instrument.ConnectionMode == FTP && instrument.FTPConfig != nil {
+		exists, err := s.instrumentRepository.WithTransaction(tx).FtpConfigExists(ctx, instrument.ID)
```

Comment:
dont update, delete the old and create a new one
also, this makes checking for existance not needed

Reply:
changed